aten/src/ATen/native/RNN.cpp

```diff
@@ -360,15 +365,63 @@ struct QuantizedCellParamsDynamic : public CellParamsBase {
         /*w_ih_packed=*/std::move(packed_params[0]),
         /*w_hh_packed=*/std::move(packed_params[1]),
         /*bias_ih=*/std::move(tensors[0]),
-        /*bias_hh=*/std::move(tensors[1]));
+        /*bias_hh=*/std::move(tensors[1]),
+        /*reduce_range=*/false);
+  }
+};
+
+struct QuantizedCellParamsDynamicReduceRange : public QuantizedCellParamsDynamic {
+  QuantizedCellParamsDynamicReduceRange(
+      c10::intrusive_ptr<LinearPackedParamsBase>
+          _packed_w_ih, /* Prepacked Weight Tensor */
+      c10::intrusive_ptr<LinearPackedParamsBase>
+          _packed_w_hh, /* Prepacked Weight Tensor */
+      Tensor _b_ih, /* float Bias Tensor */
+      Tensor _b_hh /* float Bias Tensor */)
+      : QuantizedCellParamsDynamic{_packed_w_ih,
+                                   _packed_w_hh,
+                                   _b_ih,
+                                   _b_hh,
+                                   true} {}
+
+  Tensor linear_ih(const Tensor& input_ih) const override {
+    return packed_w_ih->apply_dynamic(input_ih, /*reduce_range=*/true);
+  }
+  Tensor linear_hh(const Tensor& input_hh) const override {
+    return packed_w_hh->apply_dynamic(input_hh, /*reduce_range=*/true);
+  }
+
+  static c10::intrusive_ptr<CellParamsBase> __setstate__(
+      CellParamsSerializationType state) {
+    std::vector<at::Tensor> tensors;
+    std::vector<c10::intrusive_ptr<LinearPackedParamsBase>> packed_params;
+    std::tie(std::ignore, tensors, std::ignore, std::ignore, packed_params) =
+        std::move(state);
+    TORCH_INTERNAL_ASSERT(tensors.size() == 2);
+    TORCH_INTERNAL_ASSERT(packed_params.size() == 2);
+    return make_quantized_cell_params_dynamic(
+        /*w_ih_packed=*/std::move(packed_params[0]),
+        /*w_hh_packed=*/std::move(packed_params[1]),
+        /*bias_ih=*/std::move(tensors[0]),
+        /*bias_hh=*/std::move(tensors[1]),
+        /*reduce_range=*/true);
```

Comment:
cc @jamesr66a : Is there a better way to handle versioning for serializing custom classes?

Reply:
How about, instead of creating a new cell type, add the bool value for `reduce_range` in the `vector<int64_t>` field of `CellParamsSerializationType`. That is currently unused, but we can define a convention for `QuantizedCellParamsDynamic` that the 0-th element (if present) is the bool value for `reduce_range`. If there are no elements in that vector, assume `false`

This we we don't need to create an entirely new cell type and increase binary size

Reply:
@supriyar proposed this option yesterday, it feels a bit hacky as we are using this field like the version number. Is there a way to define a version field (but still keep BC) in CellParamsSerializationType?